apps/bookmark/js/web_manifest_helper.js

```diff
@@ -7,6 +7,24 @@
 (function WebManifestHelper(exports) {
 
   /**
+   * It specifies whether the phone is in online or offline mode.
+   *
+   * @returns {Promise} A promise that will be fulfilled when online.
+   */
+  var onConnected = function() {
+    if (navigator.onLine) {
+      return Promise.resolve();
+    }
```

Comment:
This is waiting until device gets connection [1]. Imagine, if we are offline once we enter in this UI and we try to fetch, it will never work. So we wait until the connection is ready. If it is online, the promise will be resolver immediately as it is working right now. Otherwise if will be performed once the listener [1] will be invoked.

[1] https://github.com/mozilla-b2g/gaia/pull/28591/files#diff-b93bb857441fc2e0f2646b9d22f8a723R19